cmd/snap/cmd_help.go

```diff
@@ -20,21 +20,50 @@
 package main
 
 import (
+	"bytes"
+	"fmt"
 	"os"
 
 	"github.com/snapcore/snapd/i18n"
 
 	"github.com/jessevdk/go-flags"
 )
 
-var shortHelpHelp = i18n.G("Help")
+var shortHelpHelp = i18n.G("Show help about a command")
 var longHelpHelp = i18n.G(`
-The help command shows helpful information. Unlike this. ;-)
+The help command displays information about snap commands.
 `)
 
+// addHelp adds --help like what go-flags would do for us, but hidden
+func addHelp(parser *flags.Parser) error {
+	var help struct {
+		ShowHelp func() error `short:"h" long:"help"`
+	}
+	help.ShowHelp = func() error {
+		var buf bytes.Buffer
+		parser.WriteHelp(&buf)
+		return &flags.Error{
+			Type:    flags.ErrHelp,
+			Message: buf.String(),
+		}
+	}
+	hlp, err := parser.AddGroup("Help Options", "", &help)
+	if err != nil {
+		return err
+	}
+	parser.FindOptionByLongName("help").Description = i18n.G("Show this help message")
+	hlp.Hidden = true
+
+	return nil
+}
+
 type cmdHelp struct {
-	Manpage bool `long:"man"`
-	parser  *flags.Parser
+	Manpage    bool `long:"man"`
```

Comment:
I think we could hide this option. AFAIK it's only used during packaging for generating man pages.